win32ss/user/ntuser/csr.c

```diff
@@ -197,4 +198,83 @@ CsrClientCallServer(IN OUT PCSR_API_MESSAGE ApiMessage,
     return ApiMessage->Status;
 }
 
+/*
+ * UserSystemThreadProc
+ *
+ * Called form dedicated thread in CSRSS. RIT is started in context of this
+ * thread because it needs valid Win32 process with TEB initialized.
+ */
+DWORD UserSystemThreadProc(BOOL bRemoteProcess)
+{
+    DWORD Type;
+
+    if (!gdwPendingSystemThreads)
+    {
+        ERR("gdwPendingSystemThreads is 0!\n");
+        return 0;
+    }
+
+    /* Decide which thread this will be */
+    if (gdwPendingSystemThreads & ST_RIT)
```

Comment:
With this flags technique, what happens if you have two different apps that suddently hang at the same time?

Reply:
The idea is that we will only have one of each system threads. This means that even if two apps try to create the ghost thread, only one will be created.

Reply:
But will these two apps get the "(Not responding)" stuff and be replaced by ghost windows?

Reply:
Yes, one thread should handle all ghost windows.